diff --git a/dist/useSelector.svelte.js b/dist/useSelector.svelte.js
index 628a9cd183b65e8c561dbd2bdb7955f4552b62de..28f568782f42751f03c6a948c559666417da52ee 100644
--- a/dist/useSelector.svelte.js
+++ b/dist/useSelector.svelte.js
@@ -20,7 +20,7 @@ function defaultCompare(a, b) {
  */
 export function useSelector(source, selector = (d) => d, options = {}) {
     const compare = options.compare ?? defaultCompare;
-    let slice = $state(selector(source.get()));
+    let slice = $state.raw(selector(source.get()));
     $effect(() => {
         const unsub = source.subscribe((s) => {
             const data = selector(s);
diff --git a/src/useSelector.svelte.ts b/src/useSelector.svelte.ts
index 82a72fb16a4ec83b371139af1ae5e000cd4a4ca4..c7e085669d35e72854c009c077b24d853386dde4 100644
--- a/src/useSelector.svelte.ts
+++ b/src/useSelector.svelte.ts
@@ -35,7 +35,7 @@ export function useSelector<TState, TSelected = NoInfer<TState>>(
   options: UseSelectorOptions<TSelected> = {},
 ): { readonly current: TSelected } {
   const compare = options.compare ?? defaultCompare
-  let slice = $state(selector(source.get()))
+  let slice = $state.raw(selector(source.get()))
 
   $effect(() => {
     const unsub = source.subscribe((s) => {
